src/scenechange/mod.rs

```diff
@@ -426,10 +426,11 @@ impl<T: Pixel> SceneChangeDetector<T> {
     let lines = plane1.rows_iter().zip(plane2.rows_iter());
 
     for (l1, l2) in lines {
+      let l1 = l1.get(..plane1.cfg.width).unwrap_or(l1);
```

Comment:
`l1[..plane1.cfg.width]` produces worse code? if it does put a note to try `unwrap_unchecked()` once it is stable.

Reply:
It does seem to generate slightly worse code, as it adds the bounds check and panicking code.

I can completely remove the bounds check, but it would probably be easier to do that with just `get_unchecked` which works on stable. Also side note, the equivalent of `unwrap_unchecked` can be used on stable by doing `unwrap_or_else(|| std::hint::unreachable_unchecked())`

Reply:
I'm actually slightly on the fence about using unsafe code here to eliminate the bounds check and panic code. Given that the bounds check occurs outside of the loop, the performance hit should be minimal. I might leave the final call to @lu-zero

Reply:
If we are going to remove the unsafe code my opinion on this would be to use the original solution (`l1.get(..plane1.cfg.width).unwrap_or(l1)`), since compared to the unsafe version it only adds 4 instructions (2 cmp and 2 cmov) to the beginning of each iteration over the lines (using [] causes a lot more, plus the panicking code). It is correct because in the case that that `plane1.cfg.width > l1.len()` (which I think is impossible anyway), you wouldn't be comparing the padding.